traefik/templates/ingressclass.yaml

```
@@ -3,6 +3,8 @@
 apiVersion: networking.k8s.io/v1
   {{- else if .Capabilities.APIVersions.Has "networking.k8s.io/v1beta1/IngressClass" }}
 apiVersion: networking.k8s.io/v1beta1
+  {{- else if or (eq .Values.ingressClass.fallbackApiVersion "v1beta1") (eq .Values.ingressClass.fallbackApiVersion "v1") }}
+apiVersion: {{ printf "networking.k8s.io/%s" .Values.ingressClass.fallbackApiVersion }}
```

```suggestion
  {{- else if Values.ingressClass.fallbackApiVersion}}
apiVersion: {{ printf "networking.k8s.io/%s" .Values.ingressClass.fallbackApiVersion }}
```

Shouldn't this be fine? Why is it needed to limit this if statement to these two specific fallbackApiVersions?

---

@Ornias1993 It is always "fine" to have less validation, but a courtesy check of "did they type v1beta2 on accident" only helps and has no drawbacks.